controllers/component_image_controller.go

```diff
@@ -325,16 +330,45 @@ func (r *ComponentReconciler) reportError(ctx context.Context, component *appstu
 	return r.Client.Update(ctx, component)
 }
 
-func generateRobotAccountName(component *appstudioredhatcomv1alpha1.Component) string {
-	//TODO: replace component.Namespace with the name of the Space
-	return component.Namespace + component.Spec.Application + component.Name
+// EnsureRobotAccountSecret creates or updates robot account secret.
+func (r *ComponentReconciler) EnsureRobotAccountSecret(ctx context.Context, component *appstudioredhatcomv1alpha1.Component, robotAccount *quay.RobotAccount, secretName, imageURL string) error {
+	log := ctrllog.FromContext(ctx)
+
+	robotAccountSecret := generateSecret(component, robotAccount, secretName, imageURL)
+	robotAccountSecretKey := types.NamespacedName{Namespace: robotAccountSecret.Namespace, Name: robotAccountSecret.Name}
+	existingRobotAccountSecret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, robotAccountSecretKey, existingRobotAccountSecret); err == nil {
+		existingRobotAccountSecret.StringData = robotAccountSecret.StringData
+		if err := r.Client.Update(ctx, existingRobotAccountSecret); err != nil {
+			log.Error(err, fmt.Sprintf("failed to update robot account secret %v", robotAccountSecretKey), l.Action, l.ActionUpdate)
+			return err
+		}
+	} else {
+		if !errors.IsNotFound(err) {
+			log.Error(err, fmt.Sprintf("failed to read robot account secret %v", robotAccountSecretKey), l.Action, l.ActionView)
+			return err
+		}
+		if err := r.Client.Create(ctx, &robotAccountSecret); err != nil {
+			log.Error(err, fmt.Sprintf("error writing robot account token into Secret: %v", robotAccountSecretKey), l.Action, l.ActionAdd)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// generateRobotAccountsNames returns push and pull robot account names for the given Component
+func generateRobotAccountsNames(component *appstudioredhatcomv1alpha1.Component) (string, string) {
+	pushRobotAccountName := component.Namespace + component.Spec.Application + component.Name
+	pullRobotAccountName := pushRobotAccountName + "_pull"
```

Comment:
Will it be good to keep the name of both `pullRobotAccountName` and `pullSecretName` consistent .. using `-` or `_` same in both places

Reply:
Changed